src/Symfony/Component/Security/Http/AccessToken/OidcAccessTokenHandler.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\AccessToken;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Core\Exception\InvalidOidcUserException;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * The token handler validates and retrieves the user identifier from an OIDC server using the token.
+ *
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ */
+final class OidcAccessTokenHandler implements AccessTokenHandlerInterface
+{
+    private HttpClientInterface $oidcClient;
+    private ?LoggerInterface $logger;
+    private string $userProperty;
```

Comment:
```suggestion
    private string $claim = 'sub';
```

To be in sync with the terms used in the spec: https://openid.net/specs/openid-connect-core-1_0.html#UserInfo

Also, I suggest using using the `sub` claim that is intended for this use case and is always present (by the spec): https://openid.net/specs/openid-connect-core-1_0.html#CodeIDToken

Reply:
Also I would inline it or turn it into a constant, it must always be `sub` in order to be fully spec-compliant.

Reply:
Forcing the `sub` claim prevents using any external database for user details, such as local or external data source.

I know the best would be to store all user details in the OIDC server, but many, many projects store _some_ user details in the OIDC server and in others data sources (local database, LDAP remote server, etc.).

I would prefer to keep the claim as a class argument, and set it to `sub` as default, to keep this system as close to standard but open to project behaviors.